src/renderer.ts

```diff
@@ -203,16 +214,436 @@ export class DefaultCellRenderer implements ICellRenderer {
+    $rects.attr({
+
+      'data-index': (d, i) => i,
+
+      'width': cell_dim(total_width, cols),
+      'height': (d, i) => (context.rowHeight(i)),
+      'x': function (d) {
+
+        var prev = this.previousSibling; // One previous step information.
+        return (prev === null) ? 0 : parseFloat(d3.select(prev).attr('x')) + parseFloat(d3.select(prev).attr('width'));
+      },
+      'y': 1,
+      'fill': color
+    });
+
+    context.animated($rows).attr({
+      transform: (d, i) => 'translate(' + context.cellX(i) + ',' + context.cellY(i) + ')'
+    });
+    $rows.exit().remove();
+  }
+
+  findRow($col: d3.Selection<any>, index: number) {
+    return $col.selectAll('rect.shift[data-index="' + index + '"]');
+  }
+}
+
+class SparklineCellRenderer extends DefaultCellRenderer {
+  render($col: d3.Selection<any>, col: model.MyColumn, rows: any[], context: IRenderContext) {
+    var $rows = $col.datum(col).selectAll('g.sparklinecell').data(rows, context.rowKey);
+
+    var $rows_enter = $rows.enter().append('g').attr({
+      'class': 'sparklinecell',
+      transform: function (d, i) {
+        return 'translate(' + context.cellX(i) + ',' + context.cellPrevY(i) + ')';
+      }
+    });
+    $rows_enter.append('path').attr('class', 'sparkline');
+
+       var min = col.desc.sdomain[0], max = col.desc.sdomain[1];
+    var bits, winheight;
+
+     rows.forEach(function (d, i) {
+      var data = col.getValue(d);
+
+      bits = (data.length);
+
+      winheight = context.rowHeight(i);
+      });
+
+    var x: any = d3.scale.linear().domain([0, bits]).range([0, col.getWidth()]);
+    var y: any = y = d3.scale.linear().domain([min,max]).range([winheight,0]);
+
+
+
+    $rows.attr({
+      'data-index': function (d, i) {
+        return i;
+      }
+    });
+    $rows.select('path')
+      .attr('d', function (d, i) {
+//console.log(i,x(i));
+        var line = d3.svg.line<number>()
+          .x((d, i) => x(i))
+           .y(function (d: any, i){ return y(d);});
+        return line(col.getValue(d));
+      });
+
+    // $rows_enter.append('path').attr('class', 'sparkline')
+    //   .attr('d', (d, i) => 'M' + 0 + ',' + context.rowHeight(i) / 2 + 'L' + col.getWidth() + ',' + context.rowHeight(i) / 2)
+    //   .attr('stroke', 'red');
+    context.animated($rows).attr({
+      transform: (d, i) => 'translate(' + context.cellX(i) + ',' + context.cellY(i) + ')'
+    });
+
+    $rows.exit().remove();
+
+  }
+
+}
+
+
+class VerticalbarCellRenderer extends DefaultCellRenderer {
+  render($col: d3.Selection<any>, col: model.MyColumn, rows: any[], context: IRenderContext) {
+    var $rows = $col.datum(col).selectAll('g.verticalcell').data(rows, context.rowKey);
+    var $rows_enter = $rows.enter().append('g').attr({
+      'class': 'verticalcell',
+      transform: (d, i) => 'translate(' + context.cellX(i) + ',' + context.cellPrevY(i) + ')'
+    });
+
+    $rows.attr({
+      'data-index': (d, i) => i
+    });
+
+
+
+    var bits = [];
```

Comment:
assumes again that you are getting all the data which is not true. You can either assume that all rows have the same nested array length or you have to compute/provide it externally